src/App.jsx

```diff
@@ -1,20 +1,32 @@
-import React from 'react';
+import React from "react";
 import TodoList from "./TodoList";
-import TodoForm from "./AddTodoForm"
+import TodoForm from "./AddTodoForm";
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+  );
+
+  React.useEffect(() => {
```

Comment:
If you're using a lot of hooks, it can make your code more readable to import the hooks directly from react, rather than as methods from the entire `React` package.  It's a small thing.

`import { useState, useEffect } from "react"`